### ISSUE TITLE
Is there a way to write to disk in big chunks instead of continuously?

### ISSUE THREAD

**Issue by oltodosel** [yt-dlp/yt-dlp#8884]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I tried using `--buffer-size 100M`, but this way `yt-dlp` only holds in RAM the initial 100mb, but then it writes to disk sequentially. I want it to write in 100mb chunks in order not not hog up `IO`. Can I?

**Comment by pukkandan** [comment]
**tl;dr**: Not possible

yt-dlp does not hold any chunks in memory. It's assigned to be written to disk as soon as it is read from network. Whether it is immediately flushed to actual disk is decided by the OS.

`--buffer-size` determines how much data we attempt to read in a single chunk and so in theory, it should help. But that depends on whether the website is willing to send this much data in a single call. In case of youtube, we default to an `--http-chunk-size` of 10MB since youtube servers throttle any connections that requests a larger chunk.
